src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,214 @@
+                + "JOIN cars ON drivers.id = cars.id WHERE cars.id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllDriverStatement
+                         = connection.prepareStatement(query)) {
+            getAllDriverStatement.setLong(1, carId);
+            ResultSet resultSet = getAllDriverStatement.executeQuery();
+            List<Driver> driverList = new ArrayList<>();
+            while (resultSet.next()) {
+                driverList.add(parseDriversFromResultSet(resultSet));
+            }
+            return driverList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of car by driver id "
+                    + carId, e);
+        }
+    }
+
+    private Driver parseDriversFromResultSet(ResultSet resultSet) throws SQLException {
+        Driver newDriver = new Driver();
+        newDriver.setId(resultSet.getObject("id", Long.class));
+        newDriver.setName(resultSet.getString("name"));
+        newDriver.setLicenseNumber(resultSet.getString("license_number"));
+        return newDriver;
+
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.id, cars.model, cars.manufacturer_id, manufacturers.name, "
+                + "manufacturers.country FROM cars JOIN manufacturers"
+                + " ON cars.manufacturer_id = manufacturers.id";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a list of cars",
+                    throwable);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateCarStatement
+                         = connection.prepareStatement(query)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update a car "
+                    + car, e);
+        }
+        for (int i = 0; i < car.getDrivers().size(); i++) {
+            removeDriverFromCar(car.getDrivers().get(i), car);
+            addDriverToCar(car.getDrivers().get(i), car);
```

Comment:
What if you have 0 drivers before the `update` method call and in the passed `car` you have three drivers? How this method will work?